### ISSUE TITLE
[Ghost tables] Register part of importer

### ISSUE THREAD

**Issue by ethervoid** [CartoDB/cartodb#14725]
We need to avoid the register part of the importer to duplicate the calls to Ghost Tables so we have two options here:
- Wrap all the register logic inside a transaction so when the register phase ends ghost table is executed and the result is harmless because we've already done the register part
- Make the importer acquire the bolt lock before the register phase and release it after it. The main problem with this approach is that the lock could be already acquired by other DDL task and is released in the middle of the register phase.

The register phase starts, IIRC, with [this method](https://github.com/CartoDB/cartodb/blob/master/app/models/table_registrar.rb#L10) for the imported and [this other one](https://github.com/CartoDB/cartodb/blob/master/app/connectors/importer.rb#L78) for the connectors part.

Basically this phase move the table from the `cdb_importer` schema to the user's one (public or username) and register that table in the metadata base
